flink-clients/src/test/java/org/apache/flink/client/program/ClientConnectionTest.java

```diff
@@ -115,13 +115,13 @@ public void run() {
 
 			try {
 				// wait until the caller is successful, for at most the given time
-				long now = System.currentTimeMillis();
-				long deadline = now + MAX_DELAY;
+				long now = System.nanoTime();
+				long deadline = now + MAX_DELAY*1000000;
```

Comment:
I would suggest to express the million literal as `1_000_000` - that is easier to parse for the reader, making it easier to spot errors.